### PR TITLE
Add requester pays support

### LINKED ISSUE #148: Add a flag for `x-amz-request-payer: requester` header
### Tell us more about this new feature.

This should be a quick addition.

The `aws s3` command supports the flag `--request-payer requester` to add this header, so the same flag name and value would keep things consistent.

### PR DESCRIPTION
**Pull request by arjvik**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Comment by arjvik** [comment]
Linking to #148, since apparently titles don't get the same treatment as comments

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 1]
```diff
@@ -130,13 +132,16 @@ impl S3CrtClient {
...
+
         Ok(Self {
             allocator,
             s3_client,
             event_loop_group,
             endpoint,
             next_request_counter: AtomicU64::new(0),
             user_agent_header,
+            request_payer
         })
```
Comment (dannycjones) [inline mountpoint-s3-client/src/s3_crt_client.rs:145]: Prefer to inline this, and also to maintain trailing comma.

```suggestion
        Ok(Self {
            allocator,
            s3_client,
            event_loop_group,
            endpoint,
            next_request_counter: AtomicU64::new(0),
            user_agent_header,
            request_payer: config.request_payer,
        })
```

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 2]
```diff
@@ -164,6 +169,13 @@ impl S3CrtClient {
         message.add_header(&Header::new("accept", "application/xml"))?;
         message.add_header(&Header::new("User-Agent", &self.user_agent_header))?;
 
+        match self.request_payer {
+            Some(ref payer) => {
+                message.add_header(&Header::new("x-amz-request-payer", payer))?;
+            },
+            None => {}
+        }
```
Comment (dannycjones) [inline mountpoint-s3-client/src/s3_crt_client.rs:177]: We can make this a bit simpler as we don't care about the `None` case. [Reported by clippy check](https://github.com/awslabs/mountpoint-s3/actions/runs/4458017462/jobs/7846616113#step:6:83).

```suggestion
        if let Some(ref payer) = self.request_payer {
            message.add_header(&Header::new("x-amz-request-payer", payer))?;
        }
```

**Review by dannycjones — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution!

Added some minor style comments.

File: mountpoint-s3/src/main.rs [hunk 3]
```diff
@@ -287,6 +290,7 @@ fn mount(args: CliArgs) -> anyhow::Result<FuseSession> {
         part_size: args.part_size.map(|t| t as usize),
         endpoint,
         user_agent_prefix: Some(format!("mountpoint-s3/{}", build_info::FULL_VERSION)),
+        request_payer: args.request_payer
```
Comment (dannycjones) [inline mountpoint-s3/src/main.rs:293]: Also add trailing comma here.

```suggestion
        request_payer: args.request_payer,
```

**Comment by arjvik** [comment]
Thanks for the style comments @dannycjones - I committed all of the changes you suggested!

**Review by monthonk — CHANGES_REQUESTED** [review]
Thanks for taking a look at this feature!

Added some comments about the CLI argument.

**Review by dannycjones — COMMENTED** [review]
Seconding @monthonk's comments, otherwise LGTM.

**Comment by arjvik** [comment]
Otherwise, this should be ready to merge? @monthonk @dannycjones

File: mountpoint-s3/src/main.rs [hunk 4]
```diff
@@ -107,6 +107,9 @@ struct CliArgs {
     #[clap(long, help = "Force path-style addressing")]
     pub path_addressing: bool,
 
+    #[clap(long, help = "Value of the 'x-amz-request-payer' header sent to S3")]
```
Comment (monthonk) [inline mountpoint-s3/src/main.rs:110]: Update help text a bit, should be something like "Enable accessing to requester pays bucket".

**Review by monthonk — COMMENTED** [review]
Almost there. We need to rebase to resolve the conflict with current main branch and it should be ready.

**Comment by arjvik** [comment]
Rebase complete! @monthonk

**Review by monthonk — APPROVED** [review]
LGTM!

**Review by dannycjones — APPROVED** [review]

**Comment by dannycjones** [comment]
Merged - thanks for the contribution!

**Pull request merged**